packages/node-http-handler/src/node-http-handler.ts

```diff
@@ -175,7 +182,11 @@ export class NodeHttpHandler implements HttpHandler<NodeHttpHandlerOptions> {
       // This warning will be cancelled if the request resolves.
       socketCheckTimeoutId = setTimeout(
         () => {
-          this.socketWarningTimestamp = NodeHttpHandler.checkSocketUsage(agent, this.socketWarningTimestamp);
+          this.socketWarningTimestamp = NodeHttpHandler.checkSocketUsage(
+            agent,
+            this.socketWarningTimestamp,
+            this.config!.logger
```

Comment:
So if `config!.logger` is `undefined` or `null`, this won't throw because `logger?.warn?.(`?

Reply:
it's because config is resolved earlier in this method